// vuepress-plugin-latest-articles/src/latestArticles.js
const svgClock = `<svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-clock"><circle cx="12" cy="12" r="10"></circle><polyline points="12 6 12 12 16 14"></polyline></svg>`

const defaultTitle = {
  "en-US": "Latest Articles",
  "zh-CN": "近期文章",
}

module.exports = {
  name: "LastestArticles",
  props: {
    title: {
      type: String,
      default: defaultTitle["en-US"],
    },
    titleTag: {
      type: String,
      default: "h2",
    },
    number: {
      type: Number,
      default: 10,
    },
  },
  render(h) {
    const pages = this.$site.pages
      .filter((a) => a.frontmatter?.published !== false)
      .filter((a) => a.regularPath.startsWith(this.$page.regularPath))
      .sort((a, b) => b.lastUpdatedRaw - a.lastUpdatedRaw)
      .slice(0, this.number)

    const defalutSlot = () =>
      h("div", [
        h(
          this.titleTag,
          this.title === defaultTitle["en-US"]
            ? defaultTitle[this.$lang] ?? defaultTitle["en-US"]
            : a.title
        ),
        h(
          "ul",
          pages.map((a) =>
            h("li", [
              h(
                "div",
                {
                  style: {
                    marginRight: "20px",
                    display: "flex",
                    justifyContent: "space-between",
                  },
                },
                [
                  h(
                    "a",
                    {
                      attrs: {
                        href: a.regularPath.startsWith("/")
                          ? a.regularPath.replace("/", this.$site.base)
                          : a.regularPath,
                      },
                    },
                    a.title
                  ),
                  h(
                    "div",
                    {
                      style: {
                        display: "flex",
                        alignItems: "center",
                        fontSize: "12px",
                        color: "rgba(0,0,0,.54)",
                        fontWeight: 200,
                      },
                    },
                    [
                      h("span", {
                        style: {
                          display: "flex",
                          alignItems: "center",
                          marginRight: "5px",
                        },
                        domProps: {
                          innerHTML: svgClock,
                        },
                      }),
                      h(
                        "span",
                        new Date(a.lastUpdatedRaw).toLocaleDateString()
                      ),
                    ]
                  ),
                ]
              ),
            ])
          )
        ),
      ])

    return defalutSlot()
  },
}
